heron/healthm/src/java/com/twitter/heron/healthmgr/sensors/TrackerMetricsProvider.java

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.twitter.heron.healthmgr.sensors;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.microsoft.dhalion.api.MetricsProvider;
+import com.microsoft.dhalion.metrics.ComponentMetrics;
+import com.microsoft.dhalion.metrics.InstanceMetrics;
+
+import net.minidev.json.JSONArray;
+
+import com.twitter.heron.healthmgr.common.HealthMgrConstants;
+
+public class TrackerMetricsProvider implements MetricsProvider {
+  private static final Logger LOG = Logger.getLogger(TrackerMetricsProvider.class.getName());
+  private final WebTarget baseTarget;
+
+  private Clock clock = new Clock();
+
+  @Inject
+  public TrackerMetricsProvider(@Named(HealthMgrConstants.CONF_TRACKER_URL) String trackerURL,
+                                @Named(HealthMgrConstants.CONF_TOPOLOGY_NAME) String topologyName,
+                                @Named(HealthMgrConstants.CONF_CLUSTER) String cluster,
+                                @Named(HealthMgrConstants.CONF_ENVIRON) String environ) {
+    LOG.info("Metrics will be provided by tracker at :" + trackerURL);
+
+    Client client = ClientBuilder.newClient();
+
+    this.baseTarget = client.target(trackerURL)
+        .path("topologies/metricstimeline")
+        .queryParam("cluster", cluster)
+        .queryParam("environ", environ)
+        .queryParam("topology", topologyName);
+  }
+
+  @Override
+  public Map<String, ComponentMetrics> getComponentMetrics(String metric,
+                                                           int startTimeSec,
```

Comment:
We've been using Instant and Duration yes, but not labdas AFAIK. Seems if we're using the former we should be able to use the latter. :)